### PR TITLE
Add support for partition-tables.

### PR DESCRIPTION
**Pull request by floitsch**
## Release Notes

- **New Features**
  - Introduced an optional parameter to build partition tables in the GitHub Actions workflow.
  - Added a new job for releasing partition tables, enhancing the build process.
  - Expanded the repository's scope to include partition tables alongside envelopes.

- **Documentation**
  - Updated README.md to clarify the repository's offerings, including new sections on partition tables and variants.
  - Created a new contributing guide for adding variants, streamlining the contribution process.
  - Added detailed descriptions of various ESP32 variants in the variants README.
  - Introduced a dedicated README for partition tables, explaining their configurations and usage.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,6 +23,11 @@ on:
         required: false
         type: boolean
         default: true
+      build-tables:
```
Comment (kasperl) [inline .github/workflows/build.yml:26]: build-partitions

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,6 +23,11 @@ on:
...
+        description: Build partition tables
```
Comment (kasperl) [inline .github/workflows/build.yml:27]: Build partitions

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -192,7 +197,7 @@ jobs:
           artifacts: |
             envelopes/*
           body: |
-            Envelopes for Toit SDK ${{ github.event.inputs.toit-version }}
+            Envelopes and partition tables for Toit SDK ${{ github.event.inputs.toit-version }}
```
Comment (kasperl) [inline .github/workflows/build.yml:200]: partitions

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -371,7 +376,47 @@ jobs:
...
+      - name: Set the git version
+        run: |
+          echo "TOIT_GIT_VERSION=${{ github.event.inputs.toit-version }}" >> $GITHUB_ENV
+
+      - name: Prepare partition tables
+        shell: bash
+        run: |
+          mkdir -p build
+          shopt -s nullglob
+          for csv_file in partition-tables/*/partitions.csv; do
```
Comment (kasperl) [inline .github/workflows/build.yml:405]: Let's put these in partitions/*/partitions.csv - maybe even esp32/partitions/*/partitions.cvs.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: .github/workflows/build.yml (hunk 1 above)
Comment (floitsch) [reply .github/workflows/build.yml:26]: done.

File: .github/workflows/build.yml (hunk 2 above)
Comment (floitsch) [reply .github/workflows/build.yml:27]: done.

File: .github/workflows/build.yml (hunk 3 above)
Comment (floitsch) [reply .github/workflows/build.yml:200]: done.

File: .github/workflows/build.yml (hunk 4 above)
Comment (floitsch) [reply .github/workflows/build.yml:405]: done.

**Pull request merged**